src/components/app-state.js

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, useReducer } from "react";
+import { merge, getTimeString } from "./util";
+
+const initialState = {
+  user: null,
+  entities: {},
+  lists: {},
+  reqs: {}
+};
+
+const reducer = (state, action) => {
+  state = { ...state };
```

Comment:
Well since this mutation can't affect variables outside the scope, I think it's fine here.

```js
const mutate = obj => {
  obj.a = 1;
}
```
while the code above should definitely be considered as bad practice unless it is what required.

As far as I can remember ESLint indeed will warn in this case. But this is the reason why I didn't include ESLint in the tooling haha. IMO ESLint has pretty bad default settings.